matlab/CMakeLists.txt

```diff
@@ -273,10 +388,35 @@ if(MATLAB_BUILD_TESTS)
   # Define a test executable target. TODO: Remove the placeholder test. This is
   # just for testing GoogleTest integration.
   add_executable(placeholder_test ${CMAKE_SOURCE_DIR}/src/placeholder_test.cc)
+  add_executable(mex_util_test ${CPP_SOURCE_DIR}/arrow/matlab/mex/mex_util_test.cc)
+
   # Declare a dependency on the GTest::gtest and GTest::gtest_main IMPORTED
   # targets.
   target_link_libraries(placeholder_test GTest::gtest GTest::gtest_main)
 
+  # Declare a dependency on the GTest::gtest and GTest::gtest_main IMPORTED
+  # targets.
+  target_link_libraries(mex_util_test GTest::gtest GTest::gtest_main)
+  target_link_libraries(mex_util_test arrow_matlab)
+
+  # Include the MATLAB MEX headers.
+  target_include_directories(mex_util_test PRIVATE ${Matlab_INCLUDE_DIRS})
+  # Include the C++ source headers.
+  target_include_directories(mex_util_test PRIVATE ${CPP_SOURCE_DIR})
+
   # Add a test target.
   add_test(PlaceholderTestTarget placeholder_test)
+  add_test(CheckNumArgsTestTarget mex_util_test)
+
+  # For Windows: add the directory of libmx.dll and libmex.dll to the %PATH% for
+  # running CheckNumArgsTestTarget
+  # Note: When appending to the path using set_test_properties' ENVIRONNENT property,
+  #       make sure that we escape ';' to avoid cmake from interpreting the input as
+  #       a list of strings.
+  if(WIN32)
+    set(MATLAB_DLL_DEPENDENCIES_DIR "${Matlab_ROOT_DIR}/bin/win64")
+    set_tests_properties(CheckNumArgsTestTarget
+                         PROPERTIES ENVIRONMENT
+                                    "PATH=${MATLAB_DLL_DEPENDENCIES_DIR}\;$ENV{PATH}")
```

Comment:
How about using `%PATH%` for GTest too instead of copying GTest files?

Reply:
Thanks for this suggestion, it'll be more efficient to skip copying the GTest dlls. I will make this change.